src/types/debug-utils.ts

```diff
@@ -0,0 +1,174 @@
+import algosdk from 'algosdk'
+import * as crypto from 'crypto'
+import * as fs from 'fs'
+import * as path from 'path'
+
+export const ALGOKIT_DIR = '.algokit'
+export const SOURCES_DIR = 'sources'
+export const SOURCES_FILE = 'sources.avm.json'
+export const TRACES_FILE_EXT = '.trace.avm.json'
+export const DEBUG_TRACES_DIR = 'debug_traces'
+export const TEAL_FILE_EXT = '.teal'
+export const TEAL_SOURCEMAP_EXT = '.teal.tok.map'
+
+export interface ErrnoException extends Error {
+  errno?: number | undefined
+  code?: string | undefined
+  path?: string | undefined
+  syscall?: string | undefined
+}
+
+export interface AVMDebuggerSourceMapDict {
+  'txn-group-sources': {
+    'sourcemap-location': string
+    hash: string
+  }[]
+}
+
+export class AVMDebuggerSourceMapEntry {
+  location: string
+  programHash: string
+
+  constructor(location: string, programHash: string) {
```

Comment:
you can use the `private` or `public` keyword to prefix params in the constructor to create 'auto fields'.

eg. 
```
constructor(private location: string, private programHash: string) {
  // No need to copy the params now
}
```